public/app/features/datasources/components/picker/DataSourceDropdown.tsx

```diff
@@ -12,17 +12,20 @@ import { Button, CustomScrollbar, Icon, Input, ModalsController, Portal, useStyl
 import { DataSourceList } from './DataSourceList';
 import { DataSourceLogo, DataSourceLogoPlaceHolder } from './DataSourceLogo';
 import { DataSourceModal } from './DataSourceModal';
-import { PickerContentProps, DataSourceDrawerProps } from './types';
-import { dataSourceName } from './utils';
+import { PickerContentProps, DataSourceDropdownProps } from './types';
+import { dataSourceName, useGetDatasource } from './utils';
 
-export function DataSourceDropdown(props: DataSourceDrawerProps) {
+export function DataSourceDropdown(props: DataSourceDropdownProps) {
   const { current, onChange, ...restProps } = props;
 
   const [isOpen, setOpen] = useState(false);
   const [markerElement, setMarkerElement] = useState<HTMLInputElement | null>();
   const [selectorElement, setSelectorElement] = useState<HTMLDivElement | null>();
   const [filterTerm, setFilterTerm] = useState<string>();
 
+  const getDataSource = useGetDatasource();
+  const currentDataSourceInstanceSettings = getDataSource(current);
```

Comment:
Is there any limitation to not using the param passed to the hook directly? I would prefer to delegate this to the hook, instead of having the hook as returning a getter function.
```suggestion
  const currentDataSourceInstanceSettings = useGetDatasource(current);
```